vector_search/utils.py

```diff
@@ -214,7 +196,146 @@ def _chunk_documents(encoder, texts, metadatas):
+    )
+    certification = serialized_resource.get("certification_type", {}).get("name")
+    topics = ", ".join(topic["name"] for topic in serialized_resource.get("topics", []))
+    # process course runs
+    runs = []
+    for run in serialized_resource.get("runs", []):
+        start_date = run.get("start_date")
+        formatted_date = (
+            date_parser.parse(start_date)
+            .replace(tzinfo=datetime.UTC)
+            .strftime("%B %d, %Y")
+            if start_date
+            else ""
+        )
+        location = run.get("location") or "Online"
+        duration = run.get("duration")
+        delivery_modes = (
+            ", ".join(delivery["name"] for delivery in run.get("delivery", []))
+            or "Not specified"
+        )
+        instructors = ", ".join(
+            instructor["full_name"]
+            for instructor in run.get("instructors", [])
+            if "full_name" in instructor
+        )
+        runs.append(
+            f" - Start Date: {formatted_date}, Location: {location}, "
+            f"Duration: {duration}, Format: {delivery_modes},"
+            f" Instructors: {instructors}"
+        )
+    runs_text = "\n".join(runs) if runs else ""
+    # Extract languages
+    languages = []
+    for run in serialized_resource.get("runs", []):
+        if run.get("languages"):
+            languages.extend(run["languages"])
+    unique_languages = ", ".join(set(languages))
+    # Extract levels
+    levels = []
+    for run in serialized_resource.get("runs", []):
+        if run.get("level"):
+            levels.extend(lvl["name"] for lvl in run["level"])
+    unique_levels = ", ".join(set(levels))
+    display_info = {
+        "Course Title": title,
+        "Description": description,
+        "Offered By": offered_by,
+        "Price": price,
+        "Certification": certification,
+        "Topics": topics,
+        "Level": unique_levels,
+        "Languages": unique_languages,
+        "Course Runs": runs_text,
+    }
+    rendered_info = "Information about this course:\n"
+    for section, display_text in display_info.items():
+        if display_text:
+            if len(display_text.strip().split("\n")) > 1:
+                rendered_info += f"{section} -\n{display_text}\n"
+            else:
+                rendered_info += f"{section} - {display_text}\n"
+    return rendered_info
+
+
+def _embed_course_metadata_as_contentfile(serialized_resources):
+    """
+    Embed general course info as a document in the contentfile collection
+    """
+    client = qdrant_client()
+    encoder = dense_encoder()
+    vector_name = encoder.model_short_name()
+    metadata = []
+    ids = []
+    docs = []
+    for doc in serialized_resources:
+        readable_id = doc["readable_id"]
+        resource_vector_point_id = str(vector_point_id(readable_id))
+        ids.append(resource_vector_point_id)
+        course_info_document = generate_metadata_document(doc)
+        metadata.append(
```

Comment:
Seems like other than resource_vector_point_id, most of this can be added to generate_metadata_document (or the serializer that i think you should replace it with)